src/unix/core.c

```diff
@@ -427,6 +428,32 @@ int uv__socket(int domain, int type, int protocol) {
   return sockfd;
 }
 
+/* get a file pointer to a file in read-only and close-on-exec mode */
+FILE* uv__open_file(const char* path) {
+  int fd;
+  int r;
+
+  do
+    fd = open(path, O_RDONLY | UV__O_CLOEXEC);
+  while (fd == -1 && errno == EINTR);
+
+  if (fd == -1) {
+    /* O_CLOEXEC is not available on all platforms
+       and must be set after open in those cases */
```

Comment:
You can remember this using a static variable to avoid doing the same dance each time the function is called.